cpp/src/binaryop/compiled/struct_binary_ops.cuh

```diff
@@ -47,12 +50,39 @@ void struct_compare(mutable_column_view& out,
     });
 }
 
-void struct_lexicographic_compare(mutable_column_view& out,
-                                  column_view const& lhs,
-                                  column_view const& rhs,
-                                  bool is_lhs_scalar,
-                                  bool is_rhs_scalar,
-                                  order op_order,
-                                  bool flip_output,
-                                  rmm::cuda_stream_view stream);
+template <class BinaryOperator>
+void apply_struct_lexicographic_binop(mutable_column_view& out,
```

Comment:
This function seems inappropriately named. The current name implies it only deals with structs, but it doesn't. It's just the generic dispatch function to struct vs non-struct binop implementations.

Reply:
`apply_struct_compatible_lexicographic_binop`? I was worried about the length of the function name and couldn't come up with anything reasonable.

Reply:
Using `generate_binop_result` per Nghia's suggestion.

Reply:
How about just `apply_binary_op_impl`?